### ISSUE TITLE
Request to add two graph self-supervised methods

### ISSUE THREAD

**Issue by yyou1996** [rusty1s/pytorch_geometric#1773]
## 🚀 Feature

Request to add two graph self-supervised methods to Pytorch Geometric:
(1) heuristic designed graph self-supervised tasks (ICML'20): https://arxiv.org/abs/2006.09136;
code: https://github.com/Shen-Lab/SS-GCNs;
(2) contrastive learning as self-supervision (NeurIPS'20): https://arxiv.org/abs/2010.13902;
code: https://github.com/Shen-Lab/GraphCL.

## Motivation

Self-supervision as an emerging technique has been employed to train graph neural networks (GNNs) for more transferrable, generalizable, and robust representation learning of graphs.

**Comment by rusty1s** [comment]
Thanks, I will look into it. Feel free to contribute if you like :)

**Comment by kou18n** [comment]
@rusty1s Hello, I implemented the GraphCL model with sparse_tesnor support. I use the [PyGCL](https://github.com/PyGCL/PyGCL) code, and rewrite some code. Currently, I think PyG can not support the contrastive learning model. How should I contribute to PyG?

**Comment by rusty1s** [comment]
That sounds cool :) We are very happy to take a pull request in on that one, including both the model and an example. Let me know if you have any questions.

**Comment by SauravMaheshkar** [comment]
@kou18n I'd like to work on this again. Can I take it up?

**Comment by kou18n** [comment]
> @kou18n I'd like to work on this again. Can I take it up?

Of course. I haven't submitted the code. Thank you for your help. Maybe we can finish it together.

**Comment by SauravMaheshkar** [comment]
> > @kou18n I'd like to work on this again. Can I take it up?
> 
> Of course. I haven't submitted the code. Thank you for your help. Maybe we can finish it together.

Sounds good. I'll create a new PR and we can discuss async.

**Comment by SauravMaheshkar** [comment]
PS: totally forgot about this. If someone wants to take this up
